src/platform/api/pythonApi.ts

```diff
@@ -548,19 +548,31 @@ export class InterpreterService implements IInterpreterService {
                         .join(', ')}`
                 );
                 await Promise.all(
-                    api.environments.known.map(async (item) => {
-                        try {
-                            const env = await api.environments.resolveEnvironment(item.id);
-                            const resolved = this.trackResolvedEnvironment(env);
-                            if (resolved) {
-                                allInterpreters.push(resolved);
-                            } else {
-                                traceError(`Failed to get env details from Python API for ${item.id} without an error`);
+                    api.environments.known
+                        .filter((item) => this.isValidWorkSpaceRelatedEnvironment(item))
```

Comment:
Filter with what information we have

Reply:
This is a requirement of the new Python API, they return envs for all worksapce folders